nsxt/resource_nsxt_policy_evpn_tunnel_endpoint.go

```diff
@@ -0,0 +1,220 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/tier_0s/locale_services"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtPolicyEvpnTunnelEndpoint() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyEvpnTunnelEndpointCreate,
+		Read:   resourceNsxtPolicyEvpnTunnelEndpointRead,
+		Update: resourceNsxtPolicyEvpnTunnelEndpointUpdate,
+		Delete: resourceNsxtPolicyEvpnTunnelEndpointDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceNsxtPolicyEvpnTunnelEndpointImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":                  getNsxIDSchema(),
+			"path":                    getPathSchema(),
+			"display_name":            getDisplayNameSchema(),
+			"description":             getDescriptionSchema(),
+			"revision":                getRevisionSchema(),
+			"tag":                     getTagsSchema(),
+			"external_interface_path": getPolicyPathSchema(true, true, "Path External Interfaceon Tier0 Gateway"),
+			"edge_node_path":          getPolicyPathSchema(true, false, "Edge Node Path"),
+			"local_address": {
+				Type:         schema.TypeString,
+				Description:  "Local IPv4 IP address",
+				Required:     true,
+				ValidateFunc: validation.IsIPv4Address,
+			},
+			"mtu": getMtuSchema(),
+			"locale_service_id": {
+				Type:        schema.TypeString,
+				Description: "Id of associated Gateway Locale Service on NSX",
+				Computed:    true,
+			},
+			"gateway_id": {
+				Type:        schema.TypeString,
+				Description: "Id of associated Tier0 Gateway on NSX",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func policyEvpnTunnelEndpointPatch(d *schema.ResourceData, m interface{}, gwID string, localeServiceID string, id string) error {
+	connector := getPolicyConnector(m)
+
+	description := d.Get("description").(string)
+	displayName := d.Get("display_name").(string)
+	tags := getPolicyTagsFromSchema(d)
+	edgePath := d.Get("edge_node_path").(string)
+	mtu := int64(d.Get("mtu").(int))
+	localAddress := d.Get("local_address").(string)
+	localAddressList := []string{localAddress}
+
+	obj := model.EvpnTunnelEndpointConfig{
+		DisplayName:    &displayName,
+		Description:    &description,
+		Tags:           tags,
+		EdgePath:       &edgePath,
+		LocalAddresses: localAddressList,
+	}
+
+	if mtu > 0 {
+		obj.Mtu = &mtu
+	}
```

Comment:
Shall we raise an error instead of ignoring it?

Reply:
Hi Enhao, not sure what you mean? MTU is optional here

Reply:
I see, sorry I did not see the validation.IntAtLeast(64) for mtu, so I thought it desirable to return error upon negative input.